framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -372,10 +372,15 @@ AutomaticMortarGeneration::buildMortarSegmentMesh()
     // Reconstruct the nodal normal at xi1. This will help us
     // determine the orientation of the primary elems relative to the
     // new mortar segments.
-    Point dxyz_dxi(0);
-    for (MooseIndex(secondary_elem->n_nodes()) n = 0; n < secondary_elem->n_nodes(); ++n)
-      dxyz_dxi += Moose::fe_lagrange_1D_shape_deriv(order, n, xi1) * secondary_elem->point(n);
-    auto normal = Point(dxyz_dxi(1), -dxyz_dxi(0), 0).unit();
+    std::vector<Point> nodal_normals(secondary_elem->n_nodes());
+    for (const auto n : make_range(secondary_elem->n_nodes()))
+      nodal_normals[n] = secondary_node_to_nodal_normal.at(secondary_elem->node_ptr(n));
+    Point normal(0);
+    for (const auto n : make_range(secondary_elem->n_nodes()))
+    {
+      const auto phi = Moose::fe_lagrange_1D_shape(order, n, xi1);
+      normal += phi * nodal_normals[n];
+    }
```

Comment:
Is this fix to the normal changing any of the existing results? If it isn't, it may only play a role with highly deformed surfaces?

Reply:
This fix only changes how we determine orientation of the secondary and primary elements (e.g. a kind of "which one is on the left" kind of thing). This should only improve our ability to make that determination...it shouldn't change any of the existing determinations in our tests.

Right, this should only be a factor when we have highly deformed surfaces.